### PR TITLE
Added config.h and config.c with parsing function for nodes and orch.

### PR DESCRIPTION
**Pull request by engelmi**
Used inih for parsing. For parsing ini file use parsing_ini_file_orch or parsing_ini_file_node. Also added structure configuration_node and configuration_orch to save the ini file data.

Edit: re-opened from https://github.com/engelmi/orch/pull/27

### CODE REVIEW

**Review by sdunnagan — APPROVED** [review]

**Review by engelmi — COMMENTED** [review]
The hashmap doesn't get freed and leaks memory, currently. We should fix this.

File: src/ini/config.c [hunk 1]
```diff
@@ -0,0 +1,135 @@
+#include "config.h"
+#include "ini.h"
+
+
+int match(const char *section, const char *check_section, const char *name, const char *check_name) {
+        if (strcmp(section, check_section) == 0 && strcmp(name, check_name) == 0) {
+                return 1;
```
Comment (mkemel) [inline src/ini/config.c]: We already used stdbool.h in a couple of places, I think we should stick to it. Although it doesn't make much difference except for the sake of readability

File: src/ini/config.c [hunk 2]
```diff
@@ -0,0 +1,135 @@
+#include "config.h"
+#include "ini.h"
+
+
+int match(const char *section, const char *check_section, const char *name, const char *check_name) {
+        if (strcmp(section, check_section) == 0 && strcmp(name, check_name) == 0) {
```
Comment (mkemel) [inline src/ini/config.c]: Lets import streq() and strneq() from systemd/src/fundamental/string-util-fundamental.h and create common/string.h - replacing all strcmp() except for where it's needed for hashmap. We will definitely use it more.

File: src/ini/hashmap/hashmap.h [hunk 3]
```diff
@@ -0,0 +1,53 @@
+// Copyright 2020 Joshua J Baker. All rights reserved.
```
Comment (mkemel) [inline src/ini/hashmap/hashmap.h]: I would place hashmap implementation in the common dir.

File: src/ini/config.c [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+void free_topics_hashmap(struct hashmap *topics) {
+        size_t iter = 0;
+        void *item;
+        while (hashmap_iter(topics, &iter, &item)) {
+                topic *tp = item;
+                size_t iter_kv = 0;
+                void *item_kv;
+                int len = hashmap_count(tp->keys_and_values);
+                keyValue *free_kv[len];
+                int i = 0;
```
Comment (mkemel) [inline src/ini/config.c]: Where do we use it?

**Review by mkemel — COMMENTED** [review]
Terrific work Artiom!

File: src/ini/hashmap/hashmap.h (hunk 3 above)
Comment (ArtiomDivak) [reply src/ini/hashmap/hashmap.h]: I made the change can you take a look

File: src/ini/config.c (hunk 2 above)
Comment (ArtiomDivak) [reply src/ini/config.c]: I made the change can you take a look

File: binchihua/src/ini/config.c [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+void free_topics_hashmap(struct hashmap **topics) {
+        size_t iter = 0;
+        void *item;
+        while (hashmap_iter(*topics, &iter, &item)) {
+                topic *tp = item;
+                size_t iter_kv = 0;
+                void *item_kv;
+                int len = hashmap_count(tp->keys_and_values);
+                keyValue *free_kv[len];
+                int i = 0;
```
Comment (mkemel) [inline binchihua/src/ini/config.c]: This variable is not used

**Pull request merged**